### PR TITLE
MeterpreterCommandDependencies rubocop rule ran against first batch of modules

### PR DESCRIPTION
**Pull request by cgranleese-r7**
This PR builds upon work related to #15295.

This is the first batch of modules the new `MeterpreterCommandDependencies` rubocop rule is being ran against.

These files had been previous amended to add support for the Meterpreter command dependencies prior to this new rubocop rule. I thought it made sense to test my cop against these files first and get some feedback.

This first batch of modules includes the following files:
```
lib/msf/core/post/common.rb lib/msf/core/post/file.rb 
lib/msf/core/post/windows/accounts.rb 
lib/msf/core/post/windows/eventlog.rb 
lib/msf/core/post/windows/extapi.rb 
lib/msf/core/post/windows/file_info.rb 
lib/msf/core/post/windows/file_system.rb 
lib/msf/core/post/windows/kiwi.rb 
lib/msf/core/post/windows/ldap.rb 
lib/msf/core/post/windows/mssql.rb 
lib/msf/core/post/windows/net_api.rb 
lib/msf/core/post/windows/powershell.rb 
lib/msf/core/post/windows/priv.rb 
lib/msf/core/post/windows/process.rb 
lib/msf/core/post/windows/runas.rb 
lib/msf/core/post/windows/services.rb 
lib/msf/core/post/windows/user_profiles.rb 
lib/msf/core/post/windows/wmic.rb 
modules/post/linux/gather/gnome_keyring_dump.rb 
modules/post/windows/gather/ntds_grabber.rb
```

Command used:
```
rubocop -A --only Lint/MeterpreterCommandDependencies lib/msf/core/post/common.rb lib/msf/core/post/file.rb lib/msf/core/post/windows/accounts.rb lib/msf/core/post/windows/eventlog.rb lib/msf/core/post/windows/extapi.rb lib/msf/core/post/windows/file_info.rb lib/msf/core/post/windows/file_system.rb lib/msf/core/post/windows/kiwi.rb lib/msf/core/post/windows/ldap.rb lib/msf/core/post/windows/mssql.rb lib/msf/core/post/windows/net_api.rb lib/msf/core/post/windows/powershell.rb lib/msf/core/post/windows/priv.rb lib/msf/core/post/windows/process.rb lib/msf/core/post/windows/runas.rb lib/msf/core/post/windows/services.rb lib/msf/core/post/windows/user_profiles.rb lib/msf/core/post/windows/wmic.rb modules/post/linux/gather/gnome_keyring_dump.rb modules/post/windows/gather/ntds_grabber.rb
```

**_Just to call it out, the diff for this PR will mostly include amendments to already present Meterpreter command lists. As I'm just verifying if the cop is functioning as intended._**

## Note
As seen in the tests in #15295, there is support to alter/add sections of methods depending on various situations. e.g. 
"The cop verifies if a there is an initialise method, if no initialise present, it should be generated and appended appropriately."

I have tried to identify as many of these scenarios as possible, but if there are anymore edge-cases that may not have been covered in the tests, please call them out and I can get tests added as needed 🕵️ 

## Verification

List the steps needed to make sure this thing works

- [ ] Code review.
- [ ] Verify no api calls have been missed.

### CODE REVIEW

**Comment by jmartin-r7** [comment]
Let's also ensure we update the optionally loaded test modules.

File: lib/msf/core/post/windows/runas.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ def initialize(info = {})
         'Compat' => {
           'Meterpreter' => {
             'Commands' => %w[
-              stdapi_railgun_api*
+              stdapi_railgun_*
```
Comment (smcintyre-r7) [inline lib/msf/core/post/windows/runas.rb:22]: So this changes it to require `stdapi_railgun_memread` and `stdapi_railgun_memwrite` in addition to `stdapi_railgun_api` and `stdapi_railgun_api_multi`. I don't see where the memory related functions are needed here though. There's only two implementations of railgun right now, Windows and Python. This would mark only the Windows one as compatible since the Python one lacks the direct memory related functions.

Same comment regarding the `file_system.rb` changes from `stdapi_raigun_api*` to `stdapi_railgun_*`. The `accounts.rb` mixin on the other hand does need to remain `stdapi_railgun_*` (as it's already defined) because of the use of the `railgun.util` functions `read_data`, `read_string` and `unpack_pointer`.
